src/lib.rs

```diff
@@ -1152,6 +1167,115 @@ mod test {
         assert_eq!(format!("{:?}", grid), "[[1, 2, 3][4, 5, 6][7, 8, 9]]");
     }
 
+    #[test]
+    fn fmt_pretty_int() {
+        let grid: Grid<u8> = grid![
+            [1,2,3]
+            [4,5,6]
+            [7,8,95]
+        ];
+
+        let expected_output = 
+r#"[
+    [  1,  2,  3]
+    [  4,  5,  6]
+    [  7,  8, 95]
+]"#;
+
+        assert_eq!(expected_output, format!("{:#?}", grid));
+
+        let expected_output = 
+r#"[
+    [   1,   2,   3]
+    [   4,   5,   6]
+    [   7,   8,  95]
+]"#;
+
+        assert_eq!(expected_output, format!("{:#3?}", grid));
+    }
+
+    #[test]
+    fn fmt_pretty_float() {
+        let grid: Grid<f32> = grid![
+            [1.5,2.6,3.44]
+            [4.775,5.,6.]
+            [7.1,8.23444,95.55]
+        ];
+
+        let expected_output = 
+r#"[
+    [   1.5,   2.6,   3.4]
+    [   4.8,   5.0,   6.0]
+    [   7.1,   8.2,  95.6]
+]"#;
+
+        assert_eq!(expected_output, format!("{:#5.1?}", grid));
+
+        let expected_output = 
+r#"[
+    [  1.50000,  2.60000,  3.44000]
+    [  4.77500,  5.00000,  6.00000]
+    [  7.10000,  8.23444, 95.55000]
+]"#;
+
+        assert_eq!(expected_output, format!("{:#8.5?}", grid));
+    }
+
+    #[test]
+    fn fmt_pretty_tuple() {
+        let grid: Grid<(i32, i32)> = grid![
+            [(5,66), (432, 55)]
+            [(80, 90), (5, 6)]
+        ];
+
+        let expected_output = 
+r#"[
+    [ (        5,        66), (      432,        55)]
+    [ (       80,        90), (        5,         6)]
+]"#;
+
+        assert_eq!(expected_output, format!("{:#?}", grid));
+
+        let expected_output = 
+r#"[
+    [ (  5,  66), (432,  55)]
+    [ ( 80,  90), (  5,   6)]
+]"#;
+
+        assert_eq!(expected_output, format!("{:#3?}", grid));
+    }
+
+    #[test]
+    fn fmt_pretty_struct_derived() {
+        #[derive(Debug)]
+        struct Person {
+            _name: String,
+            _precise_age: f32,
+        }
+
+        impl Person {
+            fn new(name: &str, precise_age: f32) -> Self {
+                Person {
+                    _name: name.into(),
+                    _precise_age: precise_age,
+                }
+            }
+        }
+
+        let grid: Grid<Person> = grid![
+            [Person::new("Vic", 24.5), Person::new("Mr. Very Long Name", 1955.)]
+            [Person::new("Sam", 8.9995), Person::new("John Doe", 40.14)]
+        ];
+        
+        let expected_output = 
+r#"[
+    [ Person { _name: "Vic", _precise_age: 24.50000 }, Person { _name: "Mr. Very Long Name", _precise_age: 1955.00000 }]
+    [ Person { _name: "Sam", _precise_age: 8.99950 }, Person { _name: "John Doe", _precise_age: 40.14000 }]
+]"#;
+
+        assert_eq!(expected_output, format!("{:#5.5?}", grid));
+    }
+
```

Comment:
Thx so much for adding tests. One edge case that just came into my mind is adding a pretty format test for empty grids

Reply:
I added an empty grid test and also rearranged the other tests I added to match the format of yours.

I had `assert_eq(expected, actual)` whereas you had `assert_eq(actual, expected)` so I changed to match that format.